amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
+
+        @Override
+        public void run() {
+            try {
+                log.info("----------------------------Starting stream scale----------------------");
+                if (!scalingFactor.isEmpty()) {
+                    log.info("Starting stream scaling with params : {}", this);
+                    final DescribeStreamSummaryRequest describeStreamSummaryRequest = DescribeStreamSummaryRequest.builder().streamName(streamName).build();
+                    final DescribeStreamSummaryResponse response = client.describeStreamSummary(describeStreamSummaryRequest).get();
+
+                    int openShardCount = response.streamDescriptionSummary().openShardCount();
+                    int targetShardCount;
+                    if (scalingFactor.peek() == ReshardOptions.SPLIT) {
+                        // Split case: double the number of shards
+                        targetShardCount = (int) (openShardCount * 2.0);
+                    } else {
+                        // Merge case: half the number of shards
+                        targetShardCount = (int) (openShardCount * 0.5);
+                    }
+                    log.info("Scaling stream {} from {} shards to {} shards w/ scaling factor {}", streamName, openShardCount, targetShardCount, scalingFactor.peek());
+
+                    final UpdateShardCountRequest updateShardCountRequest = UpdateShardCountRequest.builder().streamName(streamName).targetShardCount(targetShardCount).scalingType(ScalingType.UNIFORM_SCALING).build();
+                    final UpdateShardCountResponse shardCountResponse = client.updateShardCount(updateShardCountRequest).get();
+                    log.info("Executed shard scaling request. Response Details : {}", shardCountResponse.toString());
+
+                    if (--totalRotations >= 0) {
+                        scalingFactor.offer(scalingFactor.poll());
```

Comment:
FYI: it'd be more performant, and less destructive, to keep track of a pointer/index. Once you hit the end, wraparound to the head if-and-only-if another rotation is required.

This would simplify `Queue->List`, and avoid side-effects to an object **shared** with, but not controlled/encapsulated by, this class.